File: Tecnicatura3Java/L07/JavaBeans/src/main/java/domain/Persona.java
package domain;

import java.io.Serializable;

public class Persona implements Serializable {
    
    private String nombre;
    private String apellido;

    public Persona() {} // Constructor vacío obligatorio

    public Persona(String n, String a) {
        this.nombre = n;
        this.apellido = a;
    }

    public String getNombre() {
        return this.nombre;
    }

    public String getApellido() {
        return this.apellido;
    }

    public void setNombre(String n) {
        this.nombre = n;
    }

    public void setApellido(String a) {
        this.apellido = a;
    }

    @Override
    public String toString() {
        return "Persona { nombre = " + this.nombre + ", apellido = " + this.apellido + " }";
    }
}

File: Tecnicatura3Js/L07/Persona.js
class Persona {
    static contadorP = 0;

    constructor(n, a, e) {
        this._idPersona = ++contadorP;
        this._nombre = n;
        this._apellido = a;
        this._edad = e;
    }

    get idPersona(){
        return this._idPersona;
    }

    get nombre() {
        return this._nombre;
    }

    get apellido() {
        return this._apellido;
    }

    get edad() {
        return this._edad;
    }

    set nombre(n) {
        this._nombre = n;
    }

    set apellido(a) {
        this._apellido = a;
    }

    set edad(e) {
        this._edad = e;
    }

    toString() {
        return `${this._idPersona} ${this._nombre} ${this._apellido} ${this._edad}`;
    }
}
